Dataloader Creation

KCui0327 opened the pull request.

KCui0327 committed:
Completed general data processing

KCui0327 committed:
Made changes to Ryan's Comments and Dataloader interface

# Conflicts:
#	src/dataset/dataset.py
#	src/dataset/main.py
#	src/dataset/visualize.py

KCui0327 committed:
Made changes to Ryan's Comments and Dataloader interface

RyanL123 commented on src/dataset/run.py:
```diff
@@ -0,0 +1,84 @@
+    x_train, x_val, y_train, y_val = train_test_split(
+        statements,
+        verdicts,
+        test_size=0.2,
```
Can we have this as a parameter since caller might need to control this

KCui0327 commented on src/dataset/processor/politifact.py:
```diff
@@ -25,8 +27,10 @@
-_DATA_PATH = "../../data"
-_OUTPUT_PATH = "../dataset/output"
+PROJECT_DIR = Path(__file__).resolve().parents[3]
+PROJECT_DIR = Path(__file__).resolve().parents[3]
```
Yep will resolve

RyanL123 committed:
Add CI for formatting with black (#5)

* add format ci

* format
# Conflicts:
#	src/dataset/dataset.py
#	src/dataset/main.py
#	src/dataset/visualize.py

KCui0327 commented on src/dataset/run.py:
```diff
@@ -0,0 +1,84 @@
+
+def create_loader(x, y, batch_size=32, shuffle=False, num_workers=0):
+    val_dataset = VeritasDataset(None, x, y)
+    data_loader = DataLoader(val_dataset, batch_size=batch_size, shuffle=shuffle, num_workers=num_workers)
```
transforms added to dataset creation instead

KCui0327 committed:
address ryan's comments
